docs/design/002-containerized-gadgets.md

```diff
@@ -0,0 +1,603 @@
+| xdp_cpumap/           |         |            |
+| xdp                   |         |            |
+| perf_event            |         |            |
+| lwt_in                |         |            |
+| lwt_out               |         |            |
+| lwt_xmit              |         |            |
+| lwt_seg6local         |         |            |
+| cgroup_skb/ingress    |         |            |
+| cgroup_skb/egress     |         |            |
+| cgroup/skb            |         |            |
+| cgroup/sock_create    |         |            |
+| cgroup/sock_release   |         |            |
+| cgroup/sock           |         |            |
+| cgroup/post_bind4     |         |            |
+| cgroup/post_bind6     |         |            |
+| cgroup/dev            |         |            |
+| sockops               |         |            |
+| sk_skb/stream_parser  |         |            |
+| sk_skb/stream_verdict |         |            |
+| sk_skb                |         |            |
+| sk_msg                |         |            |
+| lirc_mode2            |         |            |
+| flow_dissector        |         |            |
+| cgroup/bind4          |         |            |
+| cgroup/bind6          |         |            |
+| cgroup/connect4       |         |            |
+| cgroup/connect6       |         |            |
+| cgroup/sendmsg4       |         |            |
+| cgroup/sendmsg6       |         |            |
+| cgroup/recvmsg4       |         |            |
+| cgroup/recvmsg6       |         |            |
+| cgroup/getpeername4   |         |            |
+| cgroup/getpeername6   |         |            |
+| cgroup/getsockname4   |         |            |
+| cgroup/getsockname6   |         |            |
+| cgroup/sysctl         |         |            |
+| cgroup/getsockopt     |         |            |
+| cgroup/setsockopt     |         |            |
+| struct_ops+           |         |            |
+| sk_lookup/            |         |            |
+| seccomp               |         |            |
+| kprobe.multi          |         |            |
+| kretprobe.multi       |         |            |
+
+List of categories:
+* https://github.com/cilium/ebpf/blob/v0.10.0/elf_reader.go#L1073
+* https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/lib/bpf/libbpf.c
+
+
+## Gadgets Testing
+
+Each gadget should take care of its own testing. Inspektor Gadget will provide some framework to
+help gadget developers with this, like:
+- Integration testing:
+  https://github.com/inspektor-gadget/inspektor-gadget/blob/main/integration/helpers.go
+- Unit tests: https://github.com/inspektor-gadget/inspektor-gadget/tree/main/internal/test
+
+
+Inspektor Gadget will use some specific design gadgets for testing purposes to test that all the
+APIs and functionality it provides to gadgets is working fine.
+
+TODO: We need to double check this. IMO we shouldn't be testing external gadgets in Inspektor
```

Comment:
I have the same opinion. We should only test the gadgets that we maintain and provide. Also making sure that we are testing all/most of the API and features Inspektor Gadget provides